libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -0,0 +1,792 @@
+    // @User: Standard
+    AP_GROUPINFO("MASK",  1, AP_FETtecOneWire, _motor_mask, 0),
+
+    // @Param: POLES
+    // @DisplayName: Nr. electrical poles
+    // @Description: Number of motor electrical poles
+    // @Range: 2 50
+    // @RebootRequired: False
+    // @User: Standard
+    AP_GROUPINFO("POLES", 2, AP_FETtecOneWire, _pole_count, 14),
+
+    AP_GROUPEND
+};
+
+AP_FETtecOneWire *AP_FETtecOneWire::_singleton;
+
+AP_FETtecOneWire::AP_FETtecOneWire()
+{
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (_singleton != nullptr) {
+        AP_HAL::panic("AP_FETtecOneWire must be singleton");
+    }
+#endif
+    _singleton = this;
+
+    _response_length[OW_OK] = 1;
+    _response_length[OW_BL_START_FW] = 0;       // BL only
+    _response_length[OW_REQ_TYPE] = 1;
+    _response_length[OW_REQ_SN] = 12;
+    _response_length[OW_REQ_SW_VER] = 2;
+    _response_length[OW_SET_FAST_COM_LENGTH] = 1;
+    _response_length[OW_SET_TLM_TYPE] = 1;
+}
+
+/**
+  initialize the serial port, scan the OneWire bus, setup the found ESCs
+*/
+void AP_FETtecOneWire::init()
+{
+    if (!_uart_initialised) {
+        AP_SerialManager& serial_manager = AP::serialmanager();
+        _uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_FETtecOneWire, 0);
+        if (_uart) {
+            _uart->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
+            _uart->set_unbuffered_writes(true);
+            _uart->set_blocking_writes(false);
+            _uart->begin(BAUDRATE);
+            _uart_initialised = true;
+        }
+    }
+
+    if (_uart == nullptr) {
+        return;
+    }
+
+    const uint32_t now = AP_HAL::micros();
+
+    if (_scan_active < MOTOR_COUNT_MAX) {
+        if (now - _last_send_us < DELAY_TIME_US) {
+            // scan should not be done too fast, as the bootloader has some message timing requirements.
+            return;
+        } else {
+            _last_send_us = now;
+        }
+        // scan for all ESCs in OneWire bus
+        _scan_active = scan_escs();
+        return;
+    }
+
+    if (_setup_active < MOTOR_COUNT_MAX) {
+        if (now - _last_send_us < DELAY_TIME_US) {
+            // setup should not be done too fast, as the bootloader has some message timing requirements.
+            return;
+        } else {
+            _last_send_us = now;
+        }
+        if (_found_escs_count == 0) {
+            _scan_active = 0;  // no ESC has been found yet, start scanning again
+        } else {
+            // check if in bootloader, start ESCs FW if they are and prepare fast-throttle command
+            _setup_active = init_escs();
+        }
+        return;
+    }
+
+    if (_set_full_telemetry_active < MOTOR_COUNT_MAX+1) {
+        if (now - _last_send_us < DELAY_TIME_US) {
+            // telemetry mode should not be done too fast, as the bootloader has some message timing requirements.
+            return;
+        } else {
+            _last_send_us = now;
+        }

```

Comment:
@teezeefpv is this really necessary? the loop period is 2500us and DELAY_TIME_US is 700us.